python/ray/tune/tests/test_tuner_restore.py

```diff
@@ -476,25 +476,112 @@ def load_checkpoint(self, checkpoint_path):
             with open(checkpoint_path) as f:
                 self.idx = json.loads(f.read())["idx"]
 
-    with tempfile.TemporaryDirectory() as temp_dir:
-        os.environ["TUNE_RESTORE_RETRY_NUM"] = str(retry_num)
-        tag_file = os.path.join(temp_dir, "tag")
-        tuner = Tuner(
-            MockTrainable,
-            run_config=RunConfig(
-                name="tryout_restore",
-                stop={"training_iteration": 5},
-                failure_config=FailureConfig(max_failures=1),
-                checkpoint_config=CheckpointConfig(checkpoint_frequency=1),
-            ),
-            param_space={"tag_file_path": tag_file},
+    # Set environment variable just for this test
```

Comment:
it's not a huge problem but if we want to set env vars just for a specific test, we should be using `unittest.mock.patch.dict` as a context manager or a decorator, as the code restoring the old value will not be reached if we encounter an exception, and that may impact other tests and make it more tiresome to debug.

Reply:
Good point, will wrap inside a `patch` context.